src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -1051,6 +1062,13 @@ private Object decodeBasic(Channel channel, SocketAddress remoteAddress, ByteBuf
         return position;
     }
 
+    private static Date getDate(ByteBuf buf, DeviceSession deviceSession) {
```

Comment:
I would rename this to `decodeDate` or `readDate`. Because it's not only gets the date, but also increments indexes in the buffer.